Introduce an example binary useful for profiling

RCasatta opened the pull request:
Tools like `cargo bloat` don't work on libraries but on final executable.

As seen in https://github.com/rust-bitcoin/rust-miniscript/issues/585 and https://github.com/rust-bitcoin/rust-miniscript/pull/584 the parse example is used as a base to profile binary bloat.

However, since the parse example is not using all the API surface, we may have good optimization that are not recognized as such because the optimized function are not in the tree of the functions used.

For benchmarking size optimization a specific binary that ideally touch all the API surface is needed and this MR introduce it.

More coverage will be added if this seem a good idea for reviewers.

apoelstra commented:
We may want a `maintainer-tools` repo in the org, similar to bitcoin-core/bitcoin-maintainer-tools, where we could put stuff like this. I'd also like to try unifying our test.sh scripts, API checking scripts, etc etc, there.

RCasatta commented:
Certainly it is useful in other crates too, but at the moment in particular this crate and its companion elements-miniscript have high percentage of binary size consumption in downstream libraries due to the high use of generics so it makes the most sense here.

apoelstra commented:
Sorry, I didn't actually look at the diff (because I am waiting for #645 to get in so that CI will work). I didn't realize it was actually an example in the `examples/` directory. I assumed it was a whole separate crate, similar to our embedded tests in some other repos.

Yes, this should certainly stay in this repo.

My comments about wanting a maintainer-tools repo are separate -- we should have this anyway so we can deduplicate our test scripts and write "end to end" tests of the whole ecosystem.

RCasatta commented:
Rebased and added some calls:

in debug big executable is 72Mb and the second biggest example is 51Mb (41% bigger)
in release big executable is 9Mb and the second biggest example is 7.3Mb (23% bigger)

apoelstra commented on examples/big.rs:
```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: CC0-1.0
+//! This is not an example and will surely panic if executed, the purpose of this is using the 
```
In ec751fb06b7af82f20936208c377c61960f53648:

trailing space after `the`

apoelstra replied:
Ah, the formatter got this in the last commit.

apoelstra reviewed (approved):
ACK c8d3b9ac1f9ec913eb30768439f0f5f4db9a02be thanks!

apoelstra commented:
Let's get this in. It's a super useful benchmark. I am planning to give "redo the error types" a shot in the coming week and hopefully I can get this size down.
